src/utils/GetKeycloakRoles.ts

```diff
@@ -1,33 +1,27 @@
 import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
 
-/** Gets Keycloak JWT and parses it. */
-function getJWT (): any {
-  const token = sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)
-  if (token) {
-    return parseToken(token)
-  }
-  throw new Error('Error getting Keycloak token')
-}
-
-/** Decodes and parses Keycloak token. */
-function parseToken (token: string): any {
-  try {
-    const base64Url = token.split('.')[1]
-    const base64 = decodeURIComponent(window.atob(base64Url).split('').map(function (c) {
-      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
-    }).join(''))
-    return JSON.parse(base64)
-  } catch (err) {
-    throw new Error('Error parsing Keycloak token - ' + err)
-  }
-}
-
-/** Gets Keycloak roles from JWT. */
+/** Gets Keycloak roles and returns them. */
 export function GetKeycloakRoles (): Array<string> {
   const jwt = getJWT()
-  const keycloakRoles = jwt.roles
-  if (keycloakRoles && keycloakRoles.length > 0) {
-    return keycloakRoles
+  const keycloakRoles = (jwt.roles || []) as Array<string>
+  if (keycloakRoles.length < 1) throw new Error('Invalid Keycloak roles')
+  return keycloakRoles
+
+  /** Gets Keycloak JWT and parses it. */
+  function getJWT (): any {
+    // get KC token (JWT) from session storage
+    const keycloakToken = sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)
+    if (!keycloakToken) throw new Error('Error getting Keycloak token')
+
+    // decode and parse the JWT
+    try {
+      const base64Url = keycloakToken.split('.')[1]
+      const base64 = decodeURIComponent(window.atob(base64Url).split('').map(function (c) {
+        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
+      }).join(''))
+      return JSON.parse(base64)
+    } catch (error) {
+      throw new Error('Error parsing JWT - ' + error)
+    }
```

Comment:
^^ note the nested function

NB: a nested function may not have access to `this`